src/openms/include/OpenMS/DATASTRUCTURES/OSWData.h

```diff
@@ -0,0 +1,344 @@
+        {
+          return rt_right_width_;
+        }
+        /// RT difference in seconds to the expected RT
+        float getRTDelta() const
+        {
+          return rt_delta_;
+        }
+        /// this might return QVALUE_MISSING if q-value is not annotated in the OSW file
+        float getQValue() const
+        {
+          return q_value_;
+        }
+        /// get the transition ids (can be mapped to the chromatogram XICs in sqMass data)
+        const std::vector<UInt32>& getTransitionIDs() const
+        {
+          return transition_ids_;
+        }
+
+      private:
+        float rt_experimental_{ 0 }; ///< rt apex of this feature in seconds (averaged across all transitions)
+        float rt_left_width_{ 0 };   ///< rt start in seconds
+        float rt_right_width_{ 0 };  ///< rt end in seconds
+        float rt_delta_{ 0 };        ///< rt offset from expected distance
+        float q_value_{ -1 };        ///< optional Q-value from pyProphet; equals -1 if not set;
+        std::vector<UInt32> transition_ids_; /// many features will point to the same transition (but at different RT);
+    };
+
+    /**
+      @brief A peptide with a charge state
+
+      An OSWProtein has one or more OSWPeptidePrecursor's.
+
+      The OSWPeptidePrecursor contains multiple candidate features (peak groups) of type OSWPeakGroup, only one of which is usually true.
+
+    */
+    class OPENMS_DLLAPI OSWPeptidePrecursor
+    {
+      public:
+        /// just a dummy feature to allow for acceptor output values etc
+        OSWPeptidePrecursor() = default;
+        /// custom c'tor which fills all the members with data; all members are read-only
+        OSWPeptidePrecursor(const String& seq, const short charge, const bool decoy, const float precursor_mz, std::vector<OSWPeakGroup>&& features);
+        /// Copy c'tor
+        OSWPeptidePrecursor(const OSWPeptidePrecursor& rhs) = default;
+        /// assignment operator
+        OSWPeptidePrecursor& operator=(const OSWPeptidePrecursor& rhs) = default;
+        /// move c'tor
+        OSWPeptidePrecursor(OSWPeptidePrecursor&& rhs) = default;
+        /// move assignment operator
+        OSWPeptidePrecursor& operator=(OSWPeptidePrecursor&& rhs) = default;
+
+        /// the peptide sequence (incl. mods)
+        const String& getSequence() const
+        {
+          return seq_;
+        }
+        /// precursor charge
+        short getCharge() const
+        {
+          return charge_;
+        }
+        /// is this a decoy feature (from a decoy protein)
+        bool isDecoy() const
+        {
+          return decoy_;
+        }
+        /// m/z of this charged peptide
+        float getPCMz() const
+        {
+          return precursor_mz_;
+        }
+        /// candidate explanations
+        const std::vector<OSWPeakGroup>& getFeatures() const
+        {
+          return features_;
+        }
+
+      private:
+        String seq_;
+        short charge_{0};
```

Comment:
change order for better alignment?

Reply:
here order does not influence alignment.
the `short` is always padded

Reply:
total size is currently 40 and cannot be made smaller.
The order is somewhat hierachical and similar to the db schema.

Shall i change it anyway?